python-frontend/src/main/java/org/sonar/python/semantic/v2/ProjectLevelTypeTable.java

```diff
@@ -66,16 +68,26 @@ public PythonType getType(List<String> typeFqnParts) {
     var parent = (PythonType) rootModule;
     for (int i = 0; i < typeFqnParts.size(); i++) {
       var part = typeFqnParts.get(i);
+      var moduleFqnParts = IntStream.rangeClosed(0, i)
+        .mapToObj(typeFqnParts::get)
+        .toList();
       if (parent instanceof ObjectType) {
         return PythonType.UNKNOWN;
       }
+      if (parent instanceof ModuleType moduleType) {
+        TypeWrapper typeWrapper = moduleType.members().get(part);
+        if (typeWrapper instanceof LazyTypeWrapper lazyTypeWrapper && !lazyTypeWrapper.isResolved()) {
```

Comment:
Does this replace the idea of having a `submodules` field from [SONARPY-2176](https://sonarsource.atlassian.net/browse/SONARPY-2176?atlOrigin=eyJpIjoiMjgwMjE2NWVhZThiNDk3Mjg3MzkzY2Y5N2ExMGI0OGUiLCJwIjoiaiJ9)